### PR TITLE
Refactoring user_info_test using gomock

### PR DESCRIPTION
**Pull request by didierofrivia**
A follow up PR continuing the work made on https://github.com/3scale-labs/authorino/pull/69

### CODE REVIEW

**Review by guicassolato — CHANGES_REQUESTED** [review]
This opens up for several conflicts with #76. Rejecting for now so we have the chance to review that other PR first.

**Comment by didierofrivia** [comment]
As soon https://github.com/3scale-labs/authorino/pull/76 is merged I'll resolve the conflicts

File: pkg/config/metadata/user_info_test.go [hunk 1]
```diff
@@ -6,33 +6,50 @@ import (
...
 		"userinfo_endpoint": "http://%s/userinfo"
 	}`, authServerHost, authServerHost)
-)
-
-// TODO: replace with gomock
-type authCredMock struct{}
 
-func (a *authCredMock) GetCredentialsFromReq(*envoy_auth.AttributeContext_HttpRequest) (string, error) {
-	return "", nil
+type testArtifacts struct {
```
Comment (guicassolato) [inline pkg/config/metadata/user_info_test.go:29]:
```suggestion
type userInfoTestData struct {
```

File: pkg/config/metadata/user_info_test.go [hunk 2]
```diff
@@ -64,33 +80,28 @@ func TestUserInfoCall(t *testing.T) {
...
+	ta := newTestArtifacts(ctrl)
 
-	authContextMock := NewMockAuthContext(ctrl)
-	idConfEval := NewMockIdentityConfigEvaluator(ctrl)
-	oidcEvaluator, _ := identity.NewOIDC(fmt.Sprintf("http://%s", authServerHost), &authCredMock{})
-	otherOidcEvaluator, _ := identity.NewOIDC("http://wrongServer", &authCredMock{})
-	idConfEval.EXPECT().GetOIDC().Return(otherOidcEvaluator)
-	authContextMock.EXPECT().GetResolvedIdentity().Return(idConfEval, nil)
+	wrongOidc, _ := identity.NewOIDC(fmt.Sprintf("http://wrongServer"), ta.authCredMock)
+	ta.idConfEvalMock.EXPECT().GetOIDC().Return(wrongOidc)
```
Comment (guicassolato) [inline pkg/config/metadata/user_info_test.go:102]:
```suggestion
	otherOidcEvaluator, _ := identity.NewOIDC("http://other-oidc-server", ta.authCredMock)
	ta.idConfEvalMock.EXPECT().GetOIDC().Return(otherOidcEvaluator)
```

The subtle difference between "other" and "wrong" is that it wouldn't be possible to have this particular UserInfo config if there wasn't another valid OIDC config in identity. In other words, there must be 2 OIDC identity configs (both correct); it's the UserInfo metadata config that happens to be linked to one of those identity configs that is NOT the one that resolved the token.

File: pkg/config/metadata/user_info_test.go [hunk 3]
```diff
@@ -64,33 +80,28 @@ func TestUserInfoCall(t *testing.T) {
...
 	defer ctrl.Finish()
+	ta := newUserInfoTestData(ctrl)
 
-	authContextMock := NewMockAuthContext(ctrl)
-	idConfEval := NewMockIdentityConfigEvaluator(ctrl)
-	oidcEvaluator, _ := identity.NewOIDC(fmt.Sprintf("http://%s", authServerHost), &authCredMock{})
-	otherOidcEvaluator, _ := identity.NewOIDC("http://wrongServer", &authCredMock{})
-	idConfEval.EXPECT().GetOIDC().Return(otherOidcEvaluator)
-	authContextMock.EXPECT().GetResolvedIdentity().Return(idConfEval, nil)
+	otherOidcEvaluator, _ := identity.NewOIDC(fmt.Sprintf("http://wrongServer"), ta.authCredMock)
```
Comment (guicassolato) [inline pkg/config/metadata/user_info_test.go:101]: We should not use `fmt.Sprintf` without wildcard arguments. It was raising a warning before and I fixed it.

```suggestion
	otherOidcEvaluator, _ := identity.NewOIDC("http://wrongServer", ta.authCredMock)
```

**Review by guicassolato — APPROVED** [review]

**Pull request merged**